File: 11. Container With Most Water/11. Container With Most Water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    let l=0,r=height.length-1,
    res=0,tempRes=0;
    while(l<r){
        if(height[l]<height[r]){
            tempRes=(r-l)*height[l]
            if(tempRes>res){
                res=tempRes
            }
            l+=1
        }else{
            tempRes=(r-l)*height[r]
            if(tempRes>res){
                res=tempRes
            }
            r-=1
        }
    }
    return res
};

File: 11. Container With Most Water/11. Container With Most Water.java

import java.util.*;


public class Solution {
    public static int maxArea(int[] height) {
        int l=0;
        int r=height.length-1;
        int res =0;
        int tempRes = 0;
        while (l<r){
            if(height[l]<height[r]){
                tempRes = (r-l)*height[l];
                if(tempRes>res){
                    res=tempRes;
                }
                l++;
            }else{
                tempRes=(r-l)*height[r];
                if(tempRes>res){
                    res=tempRes;
                }
                r--;
            }
        }
        return res;
    }
    public static void main(String arg[]) {
        System.out.println(maxArea(new int []{2,3,4}));
    }
}
